### PR TITLE
Trim seqs

### LINKED ISSUE #9: Implement trim_seqs

### PR DESCRIPTION
**Pull request by josenavas**
Fixes #9

### CODE REVIEW

File: scripts/deblur [hunk 1]
```diff
@@ -68,18 +68,16 @@ def deblur_seqs(seqs_fp, read_error, mean_error, error_dist, indel_prob,
...
-def trim(seqs_fp, output_fp, trim_len):
+def trim(seqs_fp, output_fp, trim_length):
     """Step 1: trim FASTA sequences"""
-    with open(seqs_fp, 'U') as f:
-        seqs = trim_seqs(parse_fasta(f), trim_len)
-
-    with open(output_fp, 'w') as f:
-        for s in seqs:
-            f.write(s.to_fasta())
+    with open(seqs_fp, 'U') as in_f:
```
Comment (wasade) [inline scripts/deblur]: btw, can do the multiple files at once, see [here](http://stackoverflow.com/questions/4617034/how-can-i-open-multiple-files-using-with-open-in-python)

File: setup.py [hunk 2]
```diff
@@ -46,6 +46,7 @@
       extras_require={'test': ["nose >= 0.10.1", "pep8"],
                       'doc': ["Sphinx >= 1.2.2", "sphinx-bootstrap-theme"]},
       install_requires=['click', 'numpy >= 1.7',
-                        'scikit-bio >= 0.2.2, < 0.3.0'],
+                        'scikit-bio >= 0.2.2, < 0.3.0',
+                        'biom-format'],
```
Comment (wasade) [inline setup.py]: suggest `biom-format >= 2.1.3, < 2.2.0`

File: scripts/deblur (hunk 1 above)
Comment (josenavas) [inline scripts/deblur]: Nice! I did not know that! Thanks!

File: setup.py (hunk 2 above)
Comment (josenavas) [inline setup.py]: Thanks! Done

File: scripts/deblur [hunk 3]
```diff
@@ -68,18 +68,15 @@ def deblur_seqs(seqs_fp, read_error, mean_error, error_dist, indel_prob,
...
+def trim(seqs_fp, output_fp, trim_length):
     """Step 1: trim FASTA sequences"""
-    with open(seqs_fp, 'U') as f:
-        seqs = trim_seqs(parse_fasta(f), trim_len)
-
-    with open(output_fp, 'w') as f:
-        for s in seqs:
-            f.write(s.to_fasta())
+    with open(seqs_fp, 'U') as in_f, open(output_fp, 'w') as out_f:
+        for label, seq in trim_seqs(parse_fasta(in_f), trim_length):
```
Comment (ElDeveloper) [inline scripts/deblur]: Note that `parse_fasta` will be removed in 0.3.0 (I see setup.py delimits the versions, but still), so I would suggest using `skbio.io.read`, perhaps also in conjunction with the `skbio.io.write` function.

Not a blocker, but I thought I would say something :fallen_leaf:

File: deblur/workflow.py [hunk 4]
```diff
@@ -7,9 +7,22 @@
...
-def trim_seqs(seqs_fp, trim_len):
-    """Step 1: trim FASTA sequences to specified length"""
-    pass
+def trim_seqs(input_seqs, trim_len):
+    """Step 1: trim FASTA sequences to specified length
+
+    Parameters
+    ----------
+    input_seqs : iterable of (str, str)
+        The list of input sequences in (label, sequence) format
```
Comment (ElDeveloper) [inline deblur/workflow.py]: The `trim_len` parameter is not described here.

File: scripts/deblur (hunk 3 above)
Comment (josenavas) [inline scripts/deblur]: Agree, currently the skbio.io.read will return a skbio's BiologicalSequence, which will require a bit more though in terms of including it here; so leaving as it is for now until we can explore how the BiologicalSequence can be included here.
However, I think this is raising a concern: we will no longer have a lightweight parser on skbio that simply returns (label, seq) tuples; which some application my need as the object might be an overkill....

File: deblur/workflow.py (hunk 4 above)
Comment (josenavas) [inline deblur/workflow.py]: Thanks! added

File: scripts/deblur (hunk 3 above)
Comment (ElDeveloper) [inline scripts/deblur]: Got it, seems to be good to leave as is.

If there are performance concerns, then this is something that should
probably be discussed in the skbio mailing list.

Comment (wasade) [inline scripts/deblur]: Agree. There may not actually be significant overhead.

File: scripts/deblur [hunk 5]
```diff
@@ -68,18 +68,15 @@ def deblur_seqs(seqs_fp, read_error, mean_error, error_dist, indel_prob,
                 type=click.Path(resolve_path=True, readable=True, exists=True,
                                 file_okay=True))
 @click.argument('output_fp', required=True,
-                type=click.Path(resolve_path=True, readable=True, exists=True,
+                type=click.Path(resolve_path=True, readable=True, exists=False,
                                 file_okay=True))
 @click.option('--trim-length', '-t', required=False, type=int, default=100,
               help="Sequence trim length")
```
Comment (ekopylova) [inline scripts/deblur]: add 'show_default=True' to @click.option() so that the default value is displayed during help ..

**Comment by ekopylova** [comment]
@josenavas, the checks are passing, can we merge this?

**Comment by josenavas** [comment]
There are merge conflicts.

**Comment by ekopylova** [comment]
@josenavas could you fix them? They appear to be a quick fix: multiple definitions of the trim_seq function in workflow.py and a library declaration in test_workflow.py

**Comment by josenavas** [comment]
Merge conflicts fixed

**Pull request merged**